foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiEmbeddingResponse.java

```diff
@@ -0,0 +1,52 @@
+package com.sap.ai.sdk.foundationmodels.openai;
+
+import static lombok.AccessLevel.NONE;
+import static lombok.AccessLevel.PACKAGE;
+
+import com.google.common.annotations.Beta;
+import com.sap.ai.sdk.foundationmodels.openai.generated.model.EmbeddingsCreate200Response;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nonnull;
+import lombok.AllArgsConstructor;
+import lombok.Setter;
+import lombok.Value;
+
+/**
+ * Represents a response from the OpenAI Embedding API.
+ *
+ * <p>A high-level wrapper over the generated model class {@code EmbeddingsCreate200Response},
+ * improving API usability for common use cases, such as extracting embeddings.
+ *
+ * @since 1.4.0
+ */
+@Beta
+@Value
+@AllArgsConstructor(access = PACKAGE)
+@Setter(value = NONE)
```

Comment:
```suggestion
@AllArgsConstructor(access = PACKAGE)
```
Response don't need to be immutable

Reply:
Again consistency sake. Additionally, It provides additional methods (equals, hashcode, getters, toString).